### ISSUE TITLE
Failed to import Maven project when the artifact name is a property

### ISSUE THREAD

**Issue by jdneo** [redhat-developer/vscode-java#2386]
##### Environment
Version: 1.66.0 (user setup)
Commit: e18005f0f1b33c29e81d732535d8c0e47cafb0b5
Date: 2022-03-30T05:50:14.623Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Windows_NT x64 10.0.22000

##### Steps To Reproduce
1. Use the project Spring petclinic: https://github.com/spring-projects/spring-petclinic
2. Update the pom file to let the artifact id reference to a property
![image](https://user-images.githubusercontent.com/6193897/161364292-14c09b74-896f-479f-82b9-158188760981.png)
3. Import as a Maven project, then import failed.

##### Logs

```
!ENTRY org.eclipse.jdt.ls.core 4 0 2022-04-02 11:20:09.495
!MESSAGE Initialization failed 
!STACK 0
java.lang.IllegalArgumentException: Path for project must have only one segment.
	at org.eclipse.core.runtime.Assert.isLegal(Assert.java:66)
	at org.eclipse.core.internal.resources.WorkspaceRoot.getProject(WorkspaceRoot.java:151)
	at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager.create(ProjectConfigurationManager.java:998)
	at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager.lambda$0(ProjectConfigurationManager.java:165)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:182)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:156)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:103)
	at org.eclipse.m2e.core.internal.embedder.MavenImpl.execute(MavenImpl.java:1416)
	at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager.importProjects(ProjectConfigurationManager.java:148)
	at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager.importProjects(ProjectConfigurationManager.java:137)
	at org.eclipse.jdt.ls.core.internal.managers.MavenProjectImporter.importToWorkspace(MavenProjectImporter.java:226)
	at org.eclipse.jdt.ls.core.internal.managers.ProjectsManager.importProjects(ProjectsManager.java:124)
	at org.eclipse.jdt.ls.core.internal.managers.ProjectsManager.initializeProjects(ProjectsManager.java:107)
	at org.eclipse.jdt.ls.core.internal.handlers.InitHandler$1.runInWorkspace(InitHandler.java:244)
	at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:42)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```
##### Additional Informations

Same error can be observed in Eclipse
![image](https://user-images.githubusercontent.com/6193897/161363990-9f47088d-9b22-4882-a4b8-57a63a65b37a.png)

But it works fine in IntelliJ.

@fbricon Looks like the m2e did not resolve the property correctly.

**Comment by snjeza** [comment]
@jdneo this is an upstream m2e issue. Could you create a new issue at https://github.com/eclipse-m2e/m2e-core/issues ?

**Comment by jdneo** [comment]
Verified that this should have been fixed in 1.8.0
